internal/test_cases/exit_test_case.go

```diff
@@ -48,21 +45,17 @@ func (t ExitTestCase) Run(asserter *logged_shell_asserter.LoggedShellAsserter, s
 	// We're expecting EOF since the program should've terminated
 	if !errors.Is(readErr, shell_executable.ErrProgramExited) {
 		if readErr == nil {
-			return fmt.Errorf("Expected program to exit with %d exit code, program is still running.", t.ExpectedExitCode)
+			return fmt.Errorf("Expected program to exit, program is still running.")
 		} else if errors.Is(readErr, condition_reader.ErrConditionNotMet) {
-			return fmt.Errorf("Expected program to exit with %d exit code, program is still running.", t.ExpectedExitCode)
+			return fmt.Errorf("Expected program to exit, program is still running.")
 		} else {
 			return fmt.Errorf("Error reading output: %v", readErr)
 		}
 	}
 
-	isTerminated, exitCode := shell.WaitForTermination()
+	isTerminated, _ := shell.WaitForTermination()
 	if !isTerminated {
-		return fmt.Errorf("Expected program to exit with %d exit code, program is still running.", t.ExpectedExitCode)
-	}
-
-	if exitCode != t.ExpectedExitCode {
-		return fmt.Errorf("Expected %d as exit code, got %d", t.ExpectedExitCode, exitCode)
+		return fmt.Errorf("Expected program to exit, program is still running.")
 	}
```

Comment:
@andy1li let's still check for the exit code to match 0 (sometimes user program will exit due to an exception, and it'll look like we didn't detect that)